query/src/pipelines/transforms/transform_sink.rs

```diff
@@ -84,7 +83,8 @@ impl Processor for SinkTransform {
         tracing::debug!("executing sinks transform");
         let tbl = self
             .ctx
-            .get_catalog()
+            //.get_catalog(&self.table_info.catalog_name)
+            .get_catalog("default")? // TODO pass this in
```

Comment:
get_catalog should not give the name, we should get them by some thing, such as the query plan, or it's complex to do it.

Reply:
agree. PR is WIP,  and a TODO item is left there (line 87).

The name of catalog is planning to be passed in, maybe something like this:

https://github.com/datafuselabs/databend/blob/9fd413686832ad0654b9581a3cdf1d6ac59a773f/common/planners/src/plan_table_drop.rs#L25-L28

or  embedded in the SinkTransform::table_info

Reply:
>  and set it to the session when parsing SQL

not sure about this, a session may need to work on serval catalogs simultaneously:

`select ... from default.db.tbl, hive.db.tbl ....`

Reply:
Oh... got ur idea, for a given `sink`, it is supposed to be working on exact one `catalog`

Reply:
This is the point, maybe it helpful for us to simplify the logic.

Reply:
> agree
> 
> The name of catalog is planning to be passed in, maybe something like this:
> 
> https://github.com/datafuselabs/databend/blob/9fd413686832ad0654b9581a3cdf1d6ac59a773f/common/planners/src/plan_table_drop.rs#L25-L28
> 
> or embedded in the SinkTransform::table_info

👌